src/services/ipfs/request_writer.rs

```diff
@@ -0,0 +1,119 @@
+use bytes::Bytes;
+
+use std::io;
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+use std::sync::Arc;
+
+use futures::channel::mpsc::Sender;
+use futures::AsyncWrite;
+use futures::ready;
+use futures::Future;
+use futures::future::BoxFuture;
+use futures::Sink;
+use futures::channel::mpsc::Receiver;
+
+use crate::error::other;
+
+use super::Backend;
+
+enum State {
+  Init,
+  Sending(BoxFuture<'static, io::Result<()>>),
+}
+
+pub struct IpfsWriterFuture {
```

Comment:
Please consider use `new_http_channel` in `io_util`.

Reply:
I was looking at it previously, but I am not sure how to plug it in. I can use `Box::pin` to pass `fut` into `RequestWriter` and skip `IpfsWriterFuture` but then I get lifetime error:

![io-channel](https://user-images.githubusercontent.com/2664090/184529330-9030f79b-4c95-47c5-a690-cefd1c0336bd.png)